src/lib.rs

```diff
@@ -1118,8 +1139,14 @@ impl Buffer {
         future
     }
 
+    /// Synchronously map the buffer for reading.
+    #[cfg(feature = "sync")]
+    pub fn map_read_sync(&self, start: BufferAddress, size: BufferAddress) -> Result<BufferReadMapping, BufferAsyncErr> {
+        pollster::block_on(self.map_read(start, size))
```

Comment:
I'm not confident about this being a good idea. Mapping a buffer is async not only because it depends on client-server separation in a browser, but also because the user may be currently used by GPU. In this case, we don't want the users to block on it. I'm thinking maybe we have only the `map_read_sync` but keep the write version only async, to avoid users shooting themselves in da foots when uploading data.

Reply:
That makes sense. I'll make this change in a little while.